packages/platforms/react-native/lib/persistence.ts

```diff
@@ -0,0 +1,45 @@
+import {
+  InMemoryPersistence,
+  isObject,
+  toPersistedPayload,
+  type Persistence,
+  type PersistenceKey,
+  type PersistencePayloadMap
+} from '@bugsnag/core-performance'
+import { type AsyncStorageStatic } from '@react-native-async-storage/async-storage'
+
+const isAsyncStorage = (value: unknown): value is AsyncStorageStatic => isObject(value) && typeof value.getItem === 'function' && typeof value.setItem === 'function'
+
+export function getReactNativePersistence (): Persistence {
+  // use @react-native-async-storage/async-storage if it's installed
+  try {
+    // eslint-disable-next-line @typescript-eslint/no-var-requires
+    const asyncStorage = require('@react-native-async-storage/async-storage')
+    if (isAsyncStorage(asyncStorage)) {
+      return new ReactNativePersistence(asyncStorage)
+    }
+  } catch {}
+
+  // store items in memory if @react-native-async-storage/async-storage isn't available
+  return new InMemoryPersistence()
+}
```

Comment:
@yousif-bugsnag  updated! this now checks if asnycStorage is present and has the API we need, otherwise falls back to using InMemoryPersistence! tested on iOS simulator and working as anticipated.